static/app/components/modals/addDashboardWidgetModal.tsx

```diff
@@ -285,6 +285,13 @@ class AddDashboardWidgetModal extends React.Component<Props, State> {
         organization,
       });
 
+      // Propagate global header selection
+      Object.assign(pathQuery, {
```

Comment:
Do we need a separate step to assign these keys? Wondering if we can add them to where we instantiate `pathQuery`.
```js
      const pathQuery = {
        displayType: widgetData.displayType,
        interval: widgetData.interval,
        title: widgetData.title,
        ...queryData,
        // Propagate global header selection
        ...selection.datetime,
        project: selection.projects,
        environment: selection.environments
      };
```

Kind of looks a little wonky with the spread operators in the middle though. Let me know what you think? 🤔 

It's probably fine because the update is close to where it gets instantiated and I feel like you did that for a logical separation, if they were further apart then I think I'd be more worried about unintended mutations

Reply:
Don't need a separate assign, this should work! Updated!